### PR TITLE
Log to stdout in production by default

### PR DESCRIPTION
**Pull request by dhh**
Logging to a file doesn't make sense in production any more. You're going to run out of space, and it doesn't play well with containers. Either you log to STDOUT, and let your container setup aggregate the logs, or you'll be switching to syslogger or whatever. You won't be logging to a file in production any more.

### CODE REVIEW

File: railties/lib/rails/generators/rails/app/templates/config/environments/production.rb.tt [hunk 1]
```diff
@@ -56,9 +56,14 @@ Rails.application.configure do
...
-  # information to avoid inadvertent exposure of personally identifiable information (PII).
-  config.log_level = :info
+  # Log to STDOUT by default
+  config.logger = ActiveSupport::Logger.new(STDOUT)
+    .tap  { |logger| logger.formatter = ::Logger::Formatter.new }
+    .then { |logger| ActiveSupport::TaggedLogging.new(logger) }
+
+  # Use the lowest log level to ensure availability of diagnostic information
+  # when problems arise.
+  config.log_level = :debug
```
Comment (rafaelfranca) [inline railties/lib/rails/generators/rails/app/templates/config/environments/production.rb.tt:66]: This is a big change, so it should be in the CHANGELOG. Also I'm not sure if it is by design, so I'm calling this out. We used to have `:debug` as default, [them we make `:info` the default](https://github.com/rails/rails/commit/229fd2a02fc694b4b7756445b6647777aa94e25d). Now we are making `:debug` default again.

File: railties/lib/rails/generators/rails/app/templates/Dockerfile.tt [hunk 2]
```diff
@@ -70,8 +70,6 @@ RUN apt-get update -qq && \
 COPY --from=build /usr/local/bundle /usr/local/bundle
 COPY --from=build /rails /rails
 
-# Deployment options
-ENV RAILS_LOG_TO_STDOUT="1" \
```
[the remark this answers is not in the document]
Comment (dhh) [reply railties/lib/rails/generators/rails/app/templates/Dockerfile.tt:74]: Yeah, I know, was going to merge the other one first, then this.

File: railties/lib/rails/generators/rails/app/templates/config/environments/production.rb.tt (hunk 1 above)
[the remark this answers is not in the document]
Comment (seanpdoyle) [reply railties/lib/rails/generators/rails/app/templates/config/environments/production.rb.tt:66]: Sounds like https://github.com/rails/rails/pull/47138#discussion_r1086910610 has this covered.

[the remark this answers is not in the document]
Comment (dhh) [reply railties/lib/rails/generators/rails/app/templates/config/environments/production.rb.tt]: Actually, I do like controlling this with an ENV! But let's stick with the pattern, so we can use RAILS_LOG_LEVEL. But let's do another PR for that.

**Comment by dhh** [comment]
Unrelated CI failures.

File: railties/lib/rails/generators/rails/app/templates/config/environments/production.rb.tt (hunk 1 above)
[the remark this answers is not in the document]
Comment (dhh) [reply railties/lib/rails/generators/rails/app/templates/config/environments/production.rb.tt]: Pursued in https://github.com/rails/rails/pull/47143. Good idea!

**Pull request merged**